spec/controllers/slack_controller_spec.rb

```diff
@@ -0,0 +1,153 @@
+require 'rails_helper'
+
+describe ::DiscourseSlack::SlackController do
+  routes { ::DiscourseSlack::Engine.routes }
+
+  PLUGIN_NAME = "discourse-slack-official".freeze
+
+  before do
+    SiteSetting.slack_outbound_webhook_url = "https://hooks.slack.com/services/abcde"
+    SiteSetting.slack_enabled = true
+  end
+
+  context "when logged in" do
+    let!(:user) { log_in(:admin) }
+
+    it "default filters" do
+      expect(PluginStoreRow.where(plugin_name: PLUGIN_NAME).count).to eq(4)
+      expect(PluginStore.get(PLUGIN_NAME, "not_first_time")).to eq(true)
+      expect(PluginStore.get(PLUGIN_NAME, "legacy_migrated")).to eq(true)
+    end
+
+    context '.index' do
+      it "returns a list of filters" do
+        xhr :get, :list
+        expect(response).to be_success
+        json = ::JSON.parse(response.body)
+        expect(json['slack']).to be_present
+      end
+    end
+
+    context '.create' do
+      it "creates a filter" do
+        expect {
+          xhr :post, :edit, {channel: '#hello', category_id: 1, filter: 'follow'}
```

Comment:
Do make sure we fix the style for other spots which you've added code for as well.